src/storage-client/src/controller.rs

```diff
@@ -886,25 +890,56 @@ where
                 .await
                 .unwrap();
 
-            let (write, mut read) = persist_client
-                .open(metadata.data_shard)
+            let write = persist_client
+                .open_writer(metadata.data_shard)
                 .await
                 .expect("invalid persist usage");
 
-            // Advance the collection's `since` as requested.
-            if let Some(since) = &description.since {
-                read.downgrade_since(since).await;
-            }
+            // Construct the handle in a separate block to ensure all error paths are diverging
+            let since_handle = {
+                let mut handle = persist_client
+                    .open_critical_since(
+                        metadata.data_shard,
+                        PersistClient::CONTROLLER_CRITICAL_SINCE,
+                    )
+                    .await
+                    .expect("invalid persist usage");
+
+                let since = description
+                    .since
+                    .clone()
+                    .unwrap_or_else(|| handle.since().clone());
+
+                // We should only continue if we can fence out any other processes
+                let our_epoch = self.state.envd_epoch + 1;
+                if handle.opaque() < &our_epoch {
```

Comment:
I agree above with the `+1` being weird. I'm not convinced by the use of `<` here either.

Reply:
Well, I was about to explain my reasoning and then it didn't make sense! I'll leave the epoch as-is and make this a `<=`

Reply:
I guess there one argument for the `<` which is that if we ever fuck up and have the same epoch in two machines one will crash instead of both happily continuing.

Reply:
> have the same epoch in two machines one will crash instead of both happily continuing.

This is not something we should defend against, its a way bigger bug than the controller!